core/services/feeds/service.go

```diff
@@ -859,6 +861,15 @@ func (s *service) ApproveSpec(ctx context.Context, id int64, force bool) error {
 				if txerr != nil && !errors.Is(txerr, sql.ErrNoRows) {
 					return fmt.Errorf("failed while checking for existing ccip job: %w", txerr)
 				}
+			case job.Stream:
+				var existingJob job.Job
+				existingJob, txerr = tx.jobORM.FindJobByExternalJobID(ctx, j.ExternalJobID)
```

Comment:
It looks like on line 814 calls `jobORM.FindJobByExternalJobID` already, does this do anything different? I see that there's a `Job.StreamID` field, should we use something like `jobORM.FindJobByStreamID`?